### PR TITLE
[MMCA - 4926] - Handle scenarios for brand new cash account and no transactions for last six months

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Code changes to display the correct view for the brand new cash account page and correct view when there are no transactions in last six months

Below have been done as part of this PR

- add new test class for cash_account_no_transactions_v2.scala.html view
- add new view named cash_account_no_transactions_v2.scala.html
- add new tests for the required components and classes
- update existing tests
- refactor the CashAccountV2ViewModel and update the view
- update csv file download link to new request transactions page

**Note: -**  The method (retrieveCashTransactions) parameters in CustomsFinancialsApiConnector.scala had to kept on single line as otherwise it is making the method longer than 50 lines. Methods should be refactored in order to make it more readable but not part of this change. These parameters would be kept on separate lines once it is refactored

### CODE REVIEW

**Review by Neozxc — APPROVED** [review]

File: app/views/components/cash_account_balance.scala.html [hunk 1]
```diff
@@ -45,12 +46,12 @@
...
-        @if(showLastTransactionsHeading){
-            @h2(
-                msg = "cf.cash-account.detail.last-transactions",
-                classes="govuk-heading-m",
-                id = Some("last-transactions")
-            )
-        }
     }
 
+    @if(showBalance && displayLastSixMonthsHeading) {
```
Comment (wg-hmrc) [inline app/views/components/cash_account_balance.scala.html:51]: Why do we need to pass another boolean ?

Comment (tyagikumarvineet) [reply app/views/components/cash_account_balance.scala.html:51]: we are not passing another boolean, we are making use of existing boolean condition. It is done, as earlier it was a nested if (if inside if) and not readable especially for the view. Hence this change has been done and its just refactoring.

File: app/views/cash_account_no_transactions_v2.scala.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        linkMessage = Some(messages("cf.cash-account.how-to-use.guidance.link.text"))
+    )
+
+    @h2(
+        msg = "cf.cash-account.transactions.request.support.heading",
+        id = Some("cash-account-help-and-support-guidance-header"),
+    )
+
+    @newTabLink(
+        linkMessage = "cf.cash-account.help-and-support.link.text",
```
Comment (AdamCoops) [inline app/views/cash_account_no_transactions_v2.scala.html:66]: do links need Ids?

File: test/controllers/CashAccountV2ControllerSpec.scala [hunk 3]
```diff
@@ -17,19 +17,20 @@
...
 import play.api.i18n.{Messages, MessagesApi}
 import play.api.inject.bind
 import play.api.test.FakeRequest
-import play.api.test.Helpers._
+import play.api.test.Helpers.*
 import services.AuditingService
 import uk.gov.hmrc.http.UpstreamErrorResponse
 import utils.SpecBase
-import views.html.{cash_account_no_transactions, cash_account_no_transactions_with_balance, cash_account_transactions_not_available}
+import views.html.{cash_account_no_transactions_v2, cash_account_no_transactions_with_balance,
```
[the remark this answers is not in the document]
Comment (tyagikumarvineet) [reply test/controllers/CashAccountV2ControllerSpec.scala]: I have reformatted however its okay to keep that way as well for imports

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**